src/components/vm/snapshots/vmSnapshotsCreateModal.jsx

```diff
@@ -114,6 +112,8 @@ export class CreateSnapshotModal extends React.Component {
                         this.setState({ inProgress: false });
                         this.dialogErrorSet(_("Snapshot failed to be created"), exc.message);
                     });
+        } else {
+            this.setState({ validationError });
         }
     }
 
```

Comment:
I would akeep validationError in state, because then when the error is shown, once user adjust the error in UI, the error is immediately hidden. 
With current approach, even after user corrects incorrect snapshot name, the error message is still present until they click on "Create" again, which could make them confused.

I would recommend the approach which is done in VM creation dialog: https://github.com/cockpit-project/cockpit-machines/blob/main/src/components/create-vm-dialog/createVmDialog.jsx#L956